cli: TestLossOfQuorumRecovery failed

cli.TestLossOfQuorumRecovery [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20998338?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20998338?buildTab=artifacts#/) on master @ [ab9a17f7cc0567df7f89d066421180e198b556ea](https://github.com/cockroachdb/cockroach/commits/ab9a17f7cc0567df7f89d066421180e198b556ea):


```
=== RUN   TestLossOfQuorumRecovery
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/logTestLossOfQuorumRecovery2043435851
    test_log_scope.go:82: use -show-logs to present logs inline
[debug recover collect-info --store=/artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/store-1 /artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/node-1.json]
debug recover collect-info --store=/artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/store-1 /artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/node-1.json
Collected recovery info from:
nodes             1
stores            1
Collected info:
replicas          81
range descriptors 0
[debug recover make-plan --confirm=y --plan=/artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/recovery-plan.json /artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/node-1.json]
[debug recover apply-plan --confirm=y --store=/artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/store-1 /artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/TestLossOfQuorumRecovery2290263589/recovery-plan.json]

id	is_live	replicas	is_decommissioning	membership	is_draining
2	false	76	true	decommissioning	false
3	false	76	true	decommissioning	false
    debug_recover_loss_of_quorum_test.go:311: 
        	Error Trace:	
        	Error:      	Not equal: 
        	            	expected: "{1,4,5}"
        	            	actual  : "{1,2,3}"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-{1,4,5}
        	            	+{1,2,3}
        	Test:       	TestLossOfQuorumRecovery
        	Messages:   	Replicas after loss of quorum recovery
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8b81fbed93efb7ab499843cacefe9048/logTestLossOfQuorumRecovery2043435851
--- FAIL: TestLossOfQuorumRecovery (354.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLossOfQuorumRecovery.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58913

--- a/pkg/cli/debug_recover_loss_of_quorum_test.go
+++ b/pkg/cli/debug_recover_loss_of_quorum_test.go
@@ -173,8 +173,7 @@ func TestLossOfQuorumRecovery(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderDeadlock(t, "slow under deadlock")
-	skip.UnderStress(t, "slow under stress")
+	skip.UnderDuress(t, "slow test")
 
 	ctx := context.Background()
 	dir, cleanupFn := testutils.TempDir(t)
@@ -283,32 +282,41 @@ func TestLossOfQuorumRecovery(t *testing.T) {
 
 	require.NoError(t, runDecommissionNodeImpl(
 		ctx, adminClient, nodeDecommissionWaitNone, nodeDecommissionChecksSkip, false,
-		[]roachpb.NodeID{roachpb.NodeID(2), roachpb.NodeID(3)}, tcAfter.Server(0).NodeID()),
-		"Failed to decommission removed nodes")
+		[]roachpb.NodeID{roachpb.NodeID(2), roachpb.NodeID(3)}, tcAfter.Server(0).NodeID(),
+	))
 
 	for i := 0; i < len(tcAfter.Servers); i++ {
-		require.NoError(t, tcAfter.Servers[i].GetStores().(*kvserver.Stores).VisitStores(func(store *kvserver.Store) error {
-			store.TestingSetReplicateQueueActive(true)
+		require.NoError(t, tcAfter.Servers[i].GetStores().(*kvserver.Stores).VisitStores(func(s *kvserver.Store) error {
+			s.TestingSetReplicateQueueActive(true)
 			return nil
-		}), "Failed to activate replication queue")
+		}))
 	}
-	require.NoError(t, tcAfter.WaitForZoneConfigPropagation(),
-		"Failed to ensure zone configs are propagated")
-	require.NoError(t, tcAfter.WaitForFullReplication(), "Failed to perform full replication")
+	require.NoError(t, tcAfter.WaitForZoneConfigPropagation())
+	require.NoError(t, tcAfter.WaitForFullReplication())
+
+	// Wait until all ranges have moved to the new nodes.
+	const wantReplicas = "{1,4,5}"
+	s = sqlutils.MakeSQLRunner(tcAfter.Conns[0])
+	testutils.SucceedsWithin(t, func() error {
+		rows := s.Query(t, "select range_id, replicas from crdb_internal.ranges")
+		defer rows.Close()
+		for rows.Next() {
+			var rangeID int
+			var replicas string
+			if err := rows.Scan(&rangeID, &replicas); err != nil {
+				return err
+			} else if replicas != wantReplicas {
+				return errors.Errorf("r%d not reconfigured, replicas: %s", rangeID, replicas)
+			}
+		}
+		return rows.Err()
+	}, testutils.SucceedsSoonDuration()*2)
 
 	for i := 0; i < len(tcAfter.Servers); i++ {
 		require.NoError(t, tcAfter.Servers[i].GetStores().(*kvserver.Stores).VisitStores(func(store *kvserver.Store) error {
 			return store.ForceConsistencyQueueProcess()
 		}), "Failed to force replicas to consistency queue")
 	}
-
-	// As a validation step we will just pick one range and get its replicas to see
-	// if they were up-replicated to the new nodes.
-	s = sqlutils.MakeSQLRunner(tcAfter.Conns[0])
-	r := s.QueryRow(t, "select replicas from crdb_internal.ranges limit 1")
-	var replicas string
-	r.Scan(&replicas)
-	require.Equal(t, "{1,4,5}", replicas, "Replicas after loss of quorum recovery")
 
 	// Validate that rangelog is updated by recovery records after cluster restarts.
 	testutils.SucceedsSoon(t, func() error {

